subsys/bluetooth/host/att.c

```diff
@@ -2205,11 +2205,14 @@ int bt_att_send(struct bt_conn *conn, struct net_buf *buf)
 		return -ENOTCONN;
 	}
 
-	k_sem_take(&att->tx_sem, K_FOREVER);
-	if (!att_is_connected(att)) {
-		BT_WARN("Disconnected");
-		k_sem_give(&att->tx_sem);
-		return -ENOTCONN;
+	/* Don't use tx_sem if caller has set it own callback */
+	if (!cb) {
+		k_sem_take(&att->tx_sem, K_FOREVER);
```

Comment:
If you're not doing `k_sem_take()` when cb is set, then there's somewhere where you also shouldn't be doing `k_sem_give()` if cb was set, right? Where is that place? I don't see it modified in this PR.

Reply:
Yep, the k_sem_give is called only on the stack cbs thus they won't be called if caller has set its own callback.